static/js/message_edit.js

```diff
@@ -566,18 +590,26 @@ exports.save_message_row_edit = function (row) {
     let new_topic;
     const old_topic = message.topic;
 
+    let stream_chaged = false;
+    let new_stream_id;
+    const old_stream_id = message.stream_id;
+
     if (message.type === "stream") {
         new_topic = row.find(".message_edit_topic").val();
         topic_changed = new_topic !== old_topic && new_topic.trim() !== "";
+
+        new_stream_id = parseInt($("#select_stream_id_" + message_id).val(), 10);
```

Comment:
I guess we should be careful to ensure this works for non-admins, which requires the DOM element being present but invisible (or a change to this code); otherwise, this will throw an exception for normal users.

Reply:
Did the DOM thing for this.